ui/apps/dashboard/src/app/(dashboard)/env/[environmentSlug]/functions/[slug]/logs/(run)/[runId]/StreamDetails.tsx

```diff
@@ -76,7 +79,22 @@ export function StreamDetails({
     <div
       className={classNames('dark grid h-full text-white', event ? 'grid-cols-2' : 'grid-cols-1')}
     >
-      {event && <EventDetails event={event} />}
+      {event && (
+        <EventDetails
+          event={event}
+          codeBlockActions={[
+            {
+              label: 'Send to Dev Server',
+              title: isRunning
+                ? 'Send event payload to running Dev Server'
+                : `Dev Server is not running at ${devServerURL}`,
+              icon: <IconCloudArrowDown />,
+              onClick: () => send(event.payload),
+              disabled: !isRunning,
+            },
+          ]}
```

Comment:
Potential improvement idea: We could export this as an action as this may be re-used in multiple places, e.g.

```ts
const { codeBlockAction } = useDevServer();
```